lib/rolify/adapters/active_record/role_adapter.rb

```diff
@@ -57,7 +57,7 @@ def add(relation, role)
 
       def remove(relation, role_name, resource = nil)
         cond = { :name => role_name }
-        cond[:resource_type] = (resource.is_a?(Class) ? resource.to_s : resource.class.name) if resource
+        cond[:resource_type] = (resource.is_a?(Class) ? Rolify.base_class_for(resource).to_s : Rolify.base_class_for(resource.class).name) if resource
```

Comment:
I know you didn't write original code, but it is confusing that we do `SomeClass.to_s` and `some_instance.class.name`. I figure we should be consistent and do `SomeClass.name` and `some_instance.class.name`. 99% of the time `to_s` and `name` act the same but I could imagine someone overriding `to_s` on an AR class and having this break. I also think if you get rid of `Rolify.base_class_for` this could be simplified with:

```ruby
(resource.is_a?(Class) ? resource.polymorphic_name : resource.class.polymorphic_name) if resource
```

https://api.rubyonrails.org/classes/ActiveRecord/Inheritance/ClassMethods.html#method-i-polymorphic_name documents this method. It is what should be saved in `resource_type` without having to find the base class.

Reply:
Thank for you fedback @thijsnado! I finally came back to this PR and adapted it to use `polymorphic_name`, definately simpler like this!